backend/server/schemas/asc.py

```diff
@@ -9,13 +9,19 @@
 import backend.settings as settings
 from backend.server.models import ASC, Project, Wallet, BlockchainState
 from backend.server.utils.contract_compiler import ContractCompiler
+from backend.server.tasks import create_asc
 
 
 class ASCError(Exception):
     """Dummy class for ASC problems"""
     pass
 
 def _get_voters_for_asc(asc: ASC) -> List[str]:
+    # If the asc hasn't been deployed, we can't call any functions on it because
+    # it has no address
+    if asc.blockchain_state != BlockchainState.COMPLETED:
```

Comment:
I think I would have a slight preference of moving all this logic to models - and having the models interact with the blockchain. WDYT?
If you agree let's open a separate issue on it as code debt

Reply:
I'm not sure that's a great idea. I have this feeling that it's best for the models to only represent what's in the DB because that's what new developers will expect. Maybe we can create wrapper models to which use containment (instead of inheritance) to provide blockchain functionality. WDYT?

Reply:
that sounds liked a solid solution to me

Reply:
I'll create an issue for it

Reply:
https://github.com/MycroCoin/mycro/issues/27